rgrow/src/models/sdc1d.rs

```diff
@@ -0,0 +1,805 @@
+* to what it does)
+* - Replace all use of index for glues to WEST_GLUE_INDEX ...
+* */
+
+use std::{
+    collections::{HashMap, HashSet},
+    usize,
+};
+
+use crate::{
+    base::{Energy, Glue, GrowError, Rate, Tile},
+    canvas::{PointSafe2, PointSafeHere},
+    state::State,
+    system::{Event, NeededUpdate, System, SystemEnum, TileBondInfo},
+    tileset::{FromTileSet, ProcessedTileSet, Size},
+};
+
+use ndarray::prelude::{Array1, Array2};
+use serde::{Deserialize, Serialize};
+
+type_alias!( f64 => Strength, RatePerConc, Conc );
+
+const WEST_GLUE_INDEX: usize = 0;
+const BOTTOM_GLUE_INDEX: usize = 1;
+const EAST_GLUE_INDEX: usize = 2;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SDC {
+    /// The anchor tiles for each of the scaffolds
+    ///
+    /// To get the anchor tile of the nth scaffold, anchor_tiles.get(n)
+    pub anchor_tiles: Vec<(PointSafe2, Tile)>,
+    pub strand_names: Vec<String>,
+    pub glue_names: Vec<String>,
+    /// Colors of the scaffolds, strands can only stick if the
+    /// colors are a perfect match
+    ///
+    /// Note that this system will accept many scaffolds, thus this is a  2d array and not a 1d
+    /// array
+    pub scaffold: Array2<Glue>,
+    /// All strands in the system, they are represented by tiles
+    /// with only glue on the south, west, and east (nothing can stuck to the top of a strand)
+    // pub strands: Array1<Tile>,
+    pub strand_concentration: Array1<Conc>,
+    /// Glues of a given strand by id
+    ///
+    /// Note that the glues will be sorted in the following manner:
+    /// [
+    ///     (0) -- [left glue, bottom glue, right glue]
+    ///     ...
+    ///     (n) -- [left glue, bottom glue, right glue]
+    /// ]
+    pub glues: Array2<Glue>,
+    /// Each strand will be given a color so that it can be easily identified
+    /// when illustrated
+    pub colors: Vec<[u8; 4]>,
+    /// The (de)attachment rates will depend on this constant(for the system) value
+    pub kf: RatePerConc,
+    /// FIXME: Change this to a vector to avoid hashing time
+    ///
+    /// Set of tiles that can stick to scaffold gap with a given glue
+    pub friends_btm: HashMap<Glue, HashSet<Tile>>,
+    /// Delta G at 37 degrees C in the formula to genereate the glue strengths
+    pub delta_g_matrix: Array2<f64>,
+    /// S in the formula to geenrate the glue strengths
+    pub entropy_matrix: Array2<f64>,
+    /// Temperature of the system
+    ///
+    /// Not pub so that it cant accidentally be changed other than with the setter function
+    /// that will also recalculate energy arrays
+    temperature: f64,
+    /// The energy with which two strands will bond
+    ///
+    /// This array is indexed as follows. Given strands x and y, where x is to the west of y
+    /// (meaning that the east of x forms a bond with the west of y), the energy of said bond
+    /// is given by energy_bonds[(x, y)]
```

Comment:
We should explicitly document here whether these are negative-is-stronger or positive-is-stronger.  I would go with negative-is-stronger, like delta_g_matrix and glue_links.